File: app/src/main/java/piotr/michalkiewicz/mealplannerclient/user/model/UserAccount.java
package piotr.michalkiewicz.mealplannerclient.user.model;

import java.io.Serializable;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class UserAccount implements Serializable {

    private String id;
    private String email;
    private String name;
    private String username;
    private String password;
    private UserSettings userSettings;
}

File: app/src/main/java/piotr/michalkiewicz/mealplannerclient/user/model/UserAccount.kt
package piotr.michalkiewicz.mealplannerclient.user.model

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class UserAccount(
    @SerializedName("email") var email: String,
    @SerializedName("username") var username: String,
    @SerializedName("password") var password: String,
    @SerializedName("roles") val roles: Set<Role>,
    @SerializedName("accountType") val accountType: AccountType,
    @SerializedName("userSettings") var userSettings: UserSettings
) : Serializable

data class SingUpUserAccount(
    @SerializedName("email") var email: String,
    @SerializedName("username") var username: String,
    @SerializedName("password") var password: String
) : Serializable

enum class AccountType(val value: String) {
    TEMP("Temp"),
    NORMAL("Normal");
}

data class Role(@SerializedName("role") val role: String)